provider/resource_keycloak_realm.go

```diff
@@ -7,6 +7,56 @@ import (
 )
 
 func resourceKeycloakRealm() *schema.Resource {
+
+	otpPolicySchema := map[string]*schema.Schema{
+		"type": {
+			Type:         schema.TypeString,
+			Description:  "OTP Type, totp for Time-Based One Time Password or hotp for counter base one time password",
+			Optional:     true,
+			Default:      "totp",
+			ValidateFunc: validation.StringInSlice([]string{"totp", "hotp"}, false),
+		},
+		"algorithm": {
+			Type:         schema.TypeString,
+			Description:  "What hashing algorithm should be used to generate the OTP.",
+			Optional:     true,
+			Default:      "HmacSHA1",
+			ValidateFunc: validation.StringInSlice([]string{"HmacSHA1", "HmacSHA256", "HmacSHA512"}, false),
+		},
+		"digits": {
+			Type: schema.TypeInt,
+			Elem: &schema.Schema{
+				Type: schema.TypeInt,
+			},
+			Default:  6,
+			Optional: true,
+		},
+		"initial_counter": {
```

Comment:
I don't actually see the setting for this in the latest three Keycloak versions.  Am I missing something?

Reply:
The Option is only visible if you use a  `OTP Type:` _Counter Based_  , for the Default Type, _Time Based_, you will got the field `OTP Token Period` ( [period](https://github.com/mrparkers/terraform-provider-keycloak/blob/efa42c17c5b5e549861dfc5a6983b2f4672ed868/provider/resource_keycloak_realm.go#L50)).

Reply:
Perfect, I just missed that then.  Thanks for your help